lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -81,15 +81,15 @@ static VarDecl*
 // what already has a witness.
 static VarDecl *addImplicitDistributedActorIDProperty(
     ClassDecl *nominal) {
-  if (!nominal)
-    return nullptr;
-  if (!nominal->isDistributedActor())
+  if (!nominal || !nominal->isDistributedActor())
     return nullptr;
 
   auto &C = nominal->getASTContext();
 
   // ==== Synthesize and add 'id' property to the actor decl
   Type propertyType = getDistributedActorIDType(nominal);
+  if (!propertyType)
```

Comment:
Should this check for `|| protocoType->hasErrorType()`? I skimmed through `getDistributedActorIDType` and it seems that null check is fine but maybe add hasError too just to be safe...

Reply:
I think for type that's equivalent but let me add just to be sure 👍

Reply:
It's not equivalent unfortunatel, boolean operator on Type checks for `Ptr != 0;` only.

Reply:
Thanks for the catch, fixed now!